Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -280,22 +281,64 @@ namespace Multiplayer
             m_metricsEvent.Enqueue(bg_captureTransportPeriod, true);
         }
 
-        // If this client wants to connect on startup, wait for all the system components to activate, and then call connect.
+        // Wait for all systems to activate because allowing this server or client to host or connect.
         // Connecting too soon causes a "version mismatch" because all of the system components haven't registered their multiplayer components.
-        if (cl_connect_onstartup)
+        if (const auto settingsRegistry = AZ::SettingsRegistry::Get())
         {
-            if (const auto settingsRegistry = AZ::SettingsRegistry::Get())
-            {
-                AZ::ComponentApplicationLifecycle::RegisterHandler(
-                    *settingsRegistry,
-                    m_componentApplicationLifecycleHandler,
-                    [this](const AZ::SettingsRegistryInterface::NotifyEventArgs&)
+            AZ::ComponentApplicationLifecycle::RegisterHandler(
+                *settingsRegistry,
+                m_componentApplicationLifecycleHandler,
+                [this](const AZ::SettingsRegistryInterface::NotifyEventArgs&)
+                {
+                    // It's now safe to register and execute the "host" command
+                    m_hostConsoleCommand = AZStd::make_unique<AZ::ConsoleFunctor<MultiplayerSystemComponent, false>>(
+                        "host",
+                        "Opens a multiplayer connection as a host for other clients to connect to",
+                        AZ::ConsoleFunctorFlags::DontReplicate | AZ::ConsoleFunctorFlags::DontDuplicate,
+                        AZ::TypeId{},
+                        *this,
+                        &MultiplayerSystemComponent::HostConsoleCommand);
+
+                    AZ::Interface<AZ::IConsole>::Get()->ExecuteDeferredConsoleCommands();
+
+                    // Begin hosting or connecting if desired
+                    if (const auto console = AZ::Interface<AZ::IConsole>::Get())
```

Comment:
The AZ::IConsole is being retrieved twice here.
The `ExecuteDeferredConsoleCommands` call can be inside of the if block

Reply:
Thank you, fixed!